lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -306,6 +319,32 @@ export class PublicClientApplication {
 
     // #region Silent Flow
 
+    async ssoSilent(request: AuthenticationParameters): Promise<TokenResponse> {
+        // block the request if made from the hidden iframe
+        BrowserUtils.blockReloadInHiddenIframes();
+
+        const silentRequest: AuthenticationParameters = {
+            ...request,
+            prompt: PromptValue.NONE
+        };
+
+        // Create acquire token url
+        const navigateUrl = await this.authModule.createAcquireTokenUrl(silentRequest);
+
+        try {
+            // Create silent handler
+            const silentHandler = new SilentHandler(this.authModule, this.browserStorage, this.config.system.loadFrameTimeout);
+            // Get the frame handle for the silent request
+            const msalFrame = await silentHandler.initiateAuthRequest(navigateUrl);
+            // Monitor the window for the hash. Return the string value and close the popup when the hash is received. Default timeout is 60 seconds.
+            const hash = await silentHandler.monitorFrameForHash(msalFrame, this.config.system.iframeHashTimeout, navigateUrl);
+            // Handle response from hash string.
+            return await silentHandler.handleCodeResponse(hash);
+        } catch(e) {
+            throw e;
```

Comment:
I imagine we'll want logging, etc here in addition to throwing?

Reply:
I will add logging once we have a good design for it. Working on it this week.